plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/internal/core/auth/AuthorizationClient.java

```diff
@@ -0,0 +1,203 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.core.auth;
+
+import static org.jboss.tools.openshift.common.core.utils.URIUtils.splitFragment;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.security.KeyManagementException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.TrustManagerFactory;
+import javax.net.ssl.X509TrustManager;
+
+import org.apache.http.Header;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.conn.ssl.AllowAllHostnameVerifier;
+import org.apache.http.conn.ssl.X509HostnameVerifier;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.jboss.tools.openshift.core.auth.IAuthorizationClient;
+import org.jboss.tools.openshift.internal.core.OpenShiftCoreActivator;
+
+import com.openshift.client.NoopSSLCertificateCallback;
+import com.openshift.client.IHttpClient.ISSLCertificateCallback;
+import com.openshift.client.OpenShiftException;
+
+@SuppressWarnings("restriction")
+public class AuthorizationClient  implements IAuthorizationClient{
```

Comment:
If those authorizers are not present in osjc, how would a 3rd pary user of osjc connect and authorize? To me this very much looks like we need this in OSJC